ffmpeg: some more improvements

jopejoe1 opened the pull request:
## Description of changes

- set pkgconfig modules depending on library
- enable opengl and rtmp
- add a comment about tensorflow being disabled
- ~point changelog to the src.rev~ tested this wrongly
- make implicit dependency on nvcodec explicit
- give cuvid its own option
- ~clean up the darwin libraries~ moving this into separated pr

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

eclairevoyant committed:
bibata-cursors: move to `pkgs/by-name`

eclairevoyant committed:
bibata-cursors: 2.0.3 -> 2.0.6

updates build process based on some upstream breaking changes, and preserves pre/post-hook phases

eclairevoyant committed:
bibata-cursors: fix meta

Guanran928 committed:
material-icons: 3.0.1 -> 4.0.0

Guanran928 committed:
material-icons: skip unwanted phases

Guanran928 committed:
material-icons: rec -> finalAttrs

Guanran928 committed:
material-icons: add update script

hellwolf committed:
build-fhsenv-bubblewrap: fix fhsenv etc entries

* added more comments.
* symlink fhsenv etc entries when necessary.

TomaSajt committed:
gramps: 5.1.6 -> 5.2.0

jopejoe1 committed:
ffmpeg: make pkgConfigModules depend on included libs

jopejoe1 committed:
ffmpeg: enable opengpl on full variant

jopejoe1 committed:
ffmpeg: enable rtmp on full variant

jopejoe1 committed:
ffmpeg: add comment about why tensorflow is disabled

jopejoe1 committed:
ffmpeg: make nvidia dependencies explicit

philiptaron committed:
lib.systems: use explicit attrset instead of `rec`

This allows refactoring in the file without accidentally modifying the
public interface of the file.

Also, pull in symbols consistently from `lib` instead of `builtins`.

philiptaron committed:
lib/systems: use lib.systems.parse and lib.systems.inspect.predicates instead of re-importing

philiptaron committed:
lib/systems: inherit from lib.systems.parse in lib/systems/inspect.nix

philiptaron committed:
lib/systems: inherit from lib.systems.inspect.predicates in lib/systems/parse.nix

philiptaron committed:
Avoid top-level `with ...;` in lib/systems/parse.nix

philiptaron committed:
Avoid top-level `with ...;` in lib/systems/inspect.nix

jlesquembre committed:
openjdk22: init at 22+36

lunik1 committed:
iosevka: 28.1.0 -> 29.0.3

wegank committed:
obs-studio-plugins.obs-source-clone: 0.1.4 -> 0.1.4-unstable-2024-02-19

jopejoe1 commented:
Moved the darwin stuff out of this pr will create a separated pr for that

Atemu commented on pkgs/development/libraries/ffmpeg/generic.nix:
```diff
@@ -91,7 +91,7 @@
 , withSvg ? withFullDeps # SVG protocol
 , withSvtav1 ? withHeadlessDeps && !stdenv.isAarch64 && !stdenv.hostPlatform.isMinGW # AV1 encoder/decoder (focused on speed and correctness)
-, withTensorflow ? false # Tensorflow dnn backend support
+, withTensorflow ? false # Tensorflow dnn backend support # Increases closure size by ~390 MiB
```
If it does indeed eval without allowInsecure now and "only" adds 390MiB, we could think about enabling it.

I'd like to resolve https://github.com/NixOS/nixpkgs/issues/271863 beforehand though.

Atemu reviewed (commented):
Did you test the newly enabled features?

***

> [opengpl](https://github.com/NixOS/nixpkgs/pull/292498/commits/b7c1accda3b74dbd942a180c5f7c17a401d105b8)

The crossover I never knew I wanted :p

jopejoe1 commented on pkgs/development/libraries/ffmpeg/generic.nix:
```diff
@@ -27,20 +27,26 @@
 , withDc1394 ? withFullDeps && !stdenv.isDarwin # IIDC-1394 grabbing (ieee 1394)
 , withDrm ? withHeadlessDeps && (with stdenv; isLinux || isFreeBSD) # libdrm support
 , withFdkAac ? withFullDeps && (!withGPL || withUnfree) # Fraunhofer FDK AAC de/encoder
+, withFfnvcodec ? withHeadlessDeps && (with stdenv; !isDarwin && !isAarch32 && !hostPlatform.isRiscV && hostPlatform == buildPlatform) # dynamically linked Nvidia code
```
Renamed

jopejoe1 commented:
> Did you test the newly enabled features?

Some basic testing

> > [opengpl](https://github.com/NixOS/nixpkgs/pull/292498/commits/b7c1accda3b74dbd942a180c5f7c17a401d105b8)
> 
> The crossover I never knew I wanted :p

 :p

r-ryantm committed:
tflint-plugins.tflint-ruleset-google: 0.26.0 -> 0.27.1

AndersonTorres committed:
libisoburn: update expression

Including some extra dependencies from xorriso.

AndersonTorres committed:
xorriso: alias to libisoburn

Since, according to the original site, they are the same.

r-ryantm committed:
sqldef: 0.16.15 -> 0.17.1

r-ryantm committed:
nwg-displays: 0.3.14 -> 0.3.16

Aleksanaa committed:
waylyrics: 0.2.12 -> 0.2.13

r-ryantm committed:
beeper: 3.100.26 -> 3.101.24

Mic92 committed:
nix-ld: move to pkgs/by-name

Mic92 committed:
nix-ld: use dynamicLinker as provided by stdenv

Mic92 committed:
nix-ld-rs: init at 2024-03-23

bobby285271 committed:
xfce.libxfce4windowing: 4.19.2 -> 4.19.3

https://gitlab.xfce.org/xfce/libxfce4windowing/-/compare/libxfce4windowing-4.19.2...libxfce4windowing-4.19.3

bobby285271 committed:
budgie.budgie-desktop: Fix build with libxfce4windowing 4.19.3

Which removes xfw_(window|application)_get_id().

bobby285271 committed:
budgie.budgie-desktop: Backport a few fixes

Since we are fetching patches already, these fixes looks
good to fetch early and are already patched in Fedora.

r-ryantm committed:
hishtory: 0.277 -> 0.282

r-ryantm committed:
kor: 0.3.6 -> 0.3.7

r-ryantm committed:
cargo-mutants: 24.2.1 -> 24.3.0

r-ryantm committed:
python311Packages.pygithub: 2.2.0 -> 2.3.0

r-ryantm committed:
pict-rs: 0.5.9 -> 0.5.10

r-ryantm committed:
python312Packages.app-model: 0.2.5 -> 0.2.6

r-ryantm committed:
keycloak: 24.0.1 -> 24.0.2

r-ryantm committed:
python312Packages.rapidfuzz: 3.6.2 -> 3.7.0

r-ryantm committed:
semantic-release: 23.0.5 -> 23.0.6

r-ryantm committed:
tidal-hifi: 5.9.0 -> 5.10.0

r-ryantm committed:
python311Packages.django-auth-ldap: 4.6.0 -> 4.7.0

r-ryantm committed:
tailscale-nginx-auth: 1.58.2 -> 1.62.0

r-ryantm committed:
roadrunner: 2023.3.10 -> 2023.3.12

r-ryantm committed:
zfs-replicate: 3.2.10 -> 3.2.11

sudoforge committed:
trezor-suite: mark as broken

This change marks `trezor-suite` as broken, as the application is unable
to connect to a hardware wallet.

Ref: nixos/nixpkgs#281975

r-ryantm committed:
apache-jena: 4.10.0 -> 5.0.0

fabaff committed:
checkov: 3.2.43 -> 3.2.44

Diff: https://github.com/bridgecrewio/checkov/compare/refs/tags/3.2.43...3.2.44

Changelog: https://github.com/bridgecrewio/checkov/releases/tag/3.2.44

fabaff committed:
checkov: refactor

fabaff committed:
python312Packages.tencentcloud-sdk-python: 3.0.1114 -> 3.0.1115

Diff: https://github.com/TencentCloud/tencentcloud-sdk-python/compare/refs/tags/3.0.1114...3.0.1115

Changelog: https://github.com/TencentCloud/tencentcloud-sdk-python/blob/3.0.1115/CHANGELOG.md

fabaff committed:
python312Packages.tencentcloud-sdk-python: refactor

r-ryantm committed:
prometheus-gitlab-ci-pipelines-exporter: 0.5.6 -> 0.5.7

r-ryantm committed:
pscale: 0.185.0 -> 0.186.0

jnsgruk committed:
screenly-cli: 0.2.4 -> 0.2.5

r-ryantm committed:
trayscale: 0.10.4 -> 0.11.0

K900 committed:
Merge pull request #298897 from r-ryantm/auto-update/trayscale

trayscale: 0.10.4 -> 0.11.0

jnsgruk committed:
Merge pull request #298889 from jnsgruk/screenly-0.2.5

screenly-cli: 0.2.4 -> 0.2.5

khaneliman committed:
yabai: refactor derivation (#297334)

Currently, yabai doesn't show up in nixos package search
and, maintainers dont get notified on pull requests.
Reorganizing derivation to resolve.

bobby285271 committed:
gedit: Patch cairo usages in snippets plugin

Fixes "TypeError: Couldn't find foreign struct converter for 'cairo.Context'".

gaelreyrol committed:
frankenphp: 1.1.0 -> 1.1.2 (#297649)

r-ryantm committed:
disko: 1.4.1 -> 1.5.0

bobby285271 committed:
Merge pull request #298673 from bobby285271/upd/budgie

budgie.budgie-desktop: Use libxfce4windowing 4.19.3

K900 committed:
path-of-building.data: 2.40.1 -> 2.41.0

Diff: https://github.com/PathOfBuildingCommunity/PathOfBuilding/compare/v2.40.1...v2.41.0

domenkozar committed:
devenv: 1.0.1 -> 1.0.2

K900 committed:
Merge pull request #298917 from K900/pob-2.41.0

path-of-building.data: 2.40.1 -> 2.41.0

wegank committed:
Merge pull request #298162 from atorres1985-contrib/xorriso

Xorriso updates

wegank committed:
Merge pull request #298829 from r-ryantm/auto-update/roadrunner

roadrunner: 2023.3.10 -> 2023.3.12

IogaMaster committed:
nufmt: init at unstable-2023-09-25 (#264500)

* maintainers: add iogamaster

* nufmt: init at unstable-2023-09-25

r-ryantm committed:
sslh: 2.1.0 -> 2.1.1

wegank committed:
Merge pull request #298739 from lunik1/iosevka-29.0.3

iosevka: 28.1.0 -> 29.0.3

nixpkgs-merge-bot[bot] committed:
Merge pull request #298914 from r-ryantm/auto-update/disko

disko: 1.4.1 -> 1.5.0

Mic92 committed:
Merge pull request #298607 from Mic92/nix-ld-rs

nix-ld-rs: init at 2024-03-23

r-ryantm committed:
python312Packages.aiopvpc: 4.3.0 -> 4.3.1

r-ryantm committed:
qlog: 0.33.1 -> 0.34.0

r-ryantm committed:
python312Packages.findpython: 0.4.1 -> 0.5.1

bobby285271 committed:
Merge pull request #298908 from bobby285271/upd/gedit

gedit: Patch cairo usages in snippets plugin

mweinelt committed:
Merge pull request #291646 from TomaSajt/gramps

gramps: 5.1.6 -> 5.2.0, drop legacy builder

mweinelt committed:
Merge pull request #298942 from r-ryantm/auto-update/python312Packages.findpython

python312Packages.findpython: 0.4.1 -> 0.5.1

r-ryantm committed:
shopware-cli: 0.4.29 -> 0.4.30

wegank committed:
Merge pull request #296935 from wegank/obs-source-clone-bump

obs-studio-plugins.obs-source-clone: 0.1.4 -> 0.1.4-unstable-2024-02-19

wegank committed:
Merge pull request #298429 from Aleksanaa/waylyrics

waylyrics: 0.2.12 -> 0.2.13

fabaff committed:
python312Packages.aiopvpc: refactor

wegank committed:
Merge pull request #295440 from jlesquembre/jl/jdk22

openjdk22: init at 22+36

prusnak committed:
Merge pull request #298862 from sudoforge/master

trezor-suite: mark as broken

r-ryantm committed:
algolia-cli: 1.6.3 -> 1.6.5

r-ryantm committed:
civo: 1.0.76 -> 1.0.77

wegank committed:
Merge pull request #261532 from eclairevoyant/bibata-cursors

bibata-cursors: 2.0.3 -> 2.0.6; move to `pkgs/by-name` and fix `meta`

NickCao committed:
Merge pull request #298951 from r-ryantm/auto-update/civo

civo: 1.0.76 -> 1.0.77

NickCao committed:
Merge pull request #298950 from r-ryantm/auto-update/algolia-cli

algolia-cli: 1.6.3 -> 1.6.5

NickCao committed:
Merge pull request #298946 from r-ryantm/auto-update/shopware-cli

shopware-cli: 0.4.29 -> 0.4.30

NickCao committed:
Merge pull request #298928 from r-ryantm/auto-update/sslh

sslh: 2.1.0 -> 2.1.1

NickCao committed:
Merge pull request #298936 from r-ryantm/auto-update/python312Packages.aiopvpc

python312Packages.aiopvpc: 4.3.0 -> 4.3.1

NickCao committed:
Merge pull request #298874 from r-ryantm/auto-update/pscale

pscale: 0.185.0 -> 0.186.0

NickCao committed:
Merge pull request #298870 from r-ryantm/auto-update/prometheus-gitlab-ci-pipelines-exporter

prometheus-gitlab-ci-pipelines-exporter: 0.5.6 -> 0.5.7

NickCao committed:
Merge pull request #298868 from fabaff/tencentcloud-sdk-python-bump

python312Packages.tencentcloud-sdk-python: 3.0.1114 -> 3.0.1115

NickCao committed:
Merge pull request #298867 from fabaff/checkov-bump

checkov: 3.2.43 -> 3.2.44

wegank committed:
Merge pull request #285460 from Guanran928/material-icons

material-icons: 3.0.1 -> 4.0.0 && cleanup

NickCao committed:
Merge pull request #298824 from r-ryantm/auto-update/python311Packages.django-auth-ldap

python311Packages.django-auth-ldap: 4.6.0 -> 4.7.0

NickCao committed:
Merge pull request #298821 from r-ryantm/auto-update/tidal-hifi

tidal-hifi: 5.9.0 -> 5.10.0

wegank committed:
Merge pull request #298841 from r-ryantm/auto-update/zfs-replicate

zfs-replicate: 3.2.10 -> 3.2.11

NickCao committed:
Merge pull request #298819 from r-ryantm/auto-update/semantic-release

semantic-release: 23.0.5 -> 23.0.6

wegank committed:
Merge pull request #298827 from r-ryantm/auto-update/tailscale-nginx-auth

tailscale-nginx-auth: 1.58.2 -> 1.62.0

tomberek committed:
Merge pull request #298547 from r-ryantm/auto-update/beeper

beeper: 3.100.26 -> 3.101.24

wegank committed:
Merge pull request #298796 from r-ryantm/auto-update/kor

kor: 0.3.6 -> 0.3.7

wegank committed:
Merge pull request #298277 from r-ryantm/auto-update/nwg-displays

nwg-displays: 0.3.14 -> 0.3.16

wegank committed:
Merge pull request #298256 from r-ryantm/auto-update/sqldef

sqldef: 0.16.15 -> 0.17.1

wegank committed:
Merge pull request #297964 from r-ryantm/auto-update/tflint-plugins.tflint-ruleset-google

tflint-plugins.tflint-ruleset-google: 0.26.0 -> 0.27.1

mattpolzin committed:
bruno: 1.11.0 -> 1.12.2 (#298809)

domenkozar committed:
Merge pull request #298918 from domenkozar/devenv-1.0.2

devenv: 1.0.1 -> 1.0.2

r-ryantm committed:
cf-terraforming: 0.18.0 -> 0.19.0 (#298030)

dotlambda committed:
Merge pull request #298820 from r-ryantm/auto-update/python312Packages.rapidfuzz

python312Packages.rapidfuzz: 3.6.2 -> 3.7.0

mkg20001 committed:
Merge pull request #298940 from r-ryantm/auto-update/qlog

qlog: 0.33.1 -> 0.34.0

7c6f434c committed:
Merge pull request #298857 from r-ryantm/auto-update/apache-jena

apache-jena: 4.10.0 -> 5.0.0

Commit:
Merge pull request #286360 from hellwolf/fix-fhsenv-bubblewrap-etc-paths

build-fhsenv-bubblewrap: fix fhsenv etc entries

infinisil committed:
Merge pull request #295374 from philiptaron/issue-208242/lib.systems

lib: use explicit name imports in `lib/systems`

NickCao committed:
Merge pull request #298812 from r-ryantm/auto-update/keycloak

keycloak: 24.0.1 -> 24.0.2

NickCao committed:
Merge pull request #298811 from r-ryantm/auto-update/python312Packages.app-model

python312Packages.app-model: 0.2.5 -> 0.2.6

NickCao committed:
Merge pull request #298810 from r-ryantm/auto-update/pict-rs

pict-rs: 0.5.9 -> 0.5.10

NickCao committed:
Merge pull request #298808 from r-ryantm/auto-update/python311Packages.pygithub

python311Packages.pygithub: 2.2.0 -> 2.3.0

NickCao committed:
Merge pull request #298799 from r-ryantm/auto-update/cargo-mutants

cargo-mutants: 24.2.1 -> 24.3.0

ck3d committed:
llama-cpp: fix blasSupport (#298567)

* llama-cpp: fix blasSupport

* llama-cpp: switch from openblas to blas

NickCao committed:
Merge pull request #298794 from r-ryantm/auto-update/hishtory

hishtory: 0.277 -> 0.282

06kellyjac committed:
tracee: 0.13.1 -> 0.20.0 (#297661)

Adjust the build process
Skip some integration tests I can't fix

Commit:
Merge remote-tracking branch 'upstream/master' into staging-next
